### PR TITLE
FormatterStep now extends `AutoCloseable`, instead of having a bunch of ad-hoc cleanup methods.

### PR DESCRIPTION
**Pull request by nedtwigg**

### CODE REVIEW

**Comment by jbduncan** [comment]
LGTM! 👍

For a future PR, I wonder if `FormatterFunc` and `FormatterFunc.Closeable` can be merged together, but I'm not sure of the repercussions on the rest of the codebase...

**Review by jbduncan — APPROVED** [review]

File: CHANGES.md
Comment (Goooler) [inline CHANGES.md:1]: Unnecessary format changes in this?

**Review by Goooler — APPROVED** [review]
And we can remove 

https://github.com/diffplug/spotless/blob/0189f717ee13acdb00745bffa86c6ed4ede53ef2/lib/src/main/java/com/diffplug/spotless/FormatterFunc.java#L40-L41

File: CHANGES.md
Comment (nedtwigg) [reply CHANGES.md:1]: Geez, thanks, I didn't even notice. Must have been an editor thing, not sure which one.

**Comment by nedtwigg** [comment]
Thanks for the feedback! I agree re: `FormatterFunc.AutoCloseable`, but that will be a future PR.

**Pull request merged**